src/Components/SearchForm/SearchForm.tsx

```diff
@@ -14,32 +15,62 @@ const SearchForm = () => {
   const [searchField, setSearchField] = useState('');
   const [searchResults, setSearchResults] = useState<string[]>([]);
   const [albumsByArtist, setAlbumsByArtist] = useState<SearchedAlbumsState[]>([]);
+  const [artistSearchError, setArtistSearchError] = useState<string>('');
+  const [albumsSearchError, setAlbumsSearchError] = useState<string>('');
+
   let searchName: string = useParams().searchName!;  
   let selectedArtist: string = useParams().artistName!;
 
-  const searchArtists = () => {
+  const searchArtists = (searchName: string) => {
     fetchData(`http://ws.audioscrobbler.com/2.0/?method=artist.search&artist=${searchName}&api_key=fcf48a134034bb684aa87d0e0309a0fd
     &format=json`)
-      .then(data => {        
-        setSearchResults(formatSearchedArtists(data.results.artistmatches.artist));
+      .then(data => {     
+        if (data.results.artistmatches.artist.length === 0) {
+          throw new Error(`Looks like we don't have any artists matching that name...`);
+        } else {
+          setSearchResults(formatSearchedArtists(data.results.artistmatches.artist));
+        }
       })
       .catch(error => {
-        console.log('fetch catch error (need DOM to show as well)', error);
+        if (error.message.includes('Cannot read properties of undefined')) {
+          setArtistSearchError(`Unable to read the name you typed, please enter a validly formatted name.`);
+          setSearchResults([]);
+        } else if (error.message === `Looks like we don't have any artists matching that name...`) {
+          setArtistSearchError(error.message);
+          setSearchResults([]);
+        } else if (error.message === 'bad response') {
+          setArtistSearchError('Uh oh, looks like something went wrong in the back, please try again later.');
+          setSearchResults([]);
+        } else {
+          setArtistSearchError('Error: please contact site administrator.');
+          setSearchResults([]);
+        }
```

Comment:
This is an interesting approach to handling no search results. It's nice how you have all the "sad paths" sort of mapped out like this, easy to understand. 

What scenarios would cause `error.message` to deep-equal "bad response" or "Error: please contact site administrator."? Since you're only throwing one type of error on line 29, I guess I'm not seeing how this would catch the other types of responses for this `fetch`.

Same question for `retrieveAlbums`.